pkg/scheduler/core/generic_scheduler.go

```diff
@@ -77,28 +101,83 @@ func (g *genericScheduler) Schedule(ctx context.Context, placement *policyv1alph
 	return result, nil
 }
 
+// calcReservedClusters returns clusters that should be reserved in `binding.spec.clusters`
+func calcReservedClusters(
+	bindClusters, readyClusters, unreadyClusters sets.String,
+	candidateClusters []*clusterv1alpha1.Cluster,
+	placement *policyv1alpha1.Placement) sets.String {
+	// reservedClusters are ready clusters in `binding.spec.clusters`
+	reservedClusters := bindClusters.Intersection(readyClusters)
+	// unreadyBindClusters are "unready" clusters in `binding.spec.clusters`
+	unreadyBindClusters := bindClusters.Intersection(unreadyClusters)
+
+	if unreadyBindClusters.Len() > 0 {
+		if !features.FeatureGate.Enabled(features.Failover) {
+			// reserve unready clusters if failover is disabled
+			reservedClusters = reservedClusters.Union(unreadyBindClusters)
+		} else if len(candidateClusters) < unreadyBindClusters.Len() {
+			// *DON'T evict unready clusters in `binding.spec.clusters` if there are insufficient candidate clusters*
+			// for ReplicaSchedulingTypeDivided, we will try to migrate replicas to the other health clusters
+			if placement.ReplicaScheduling == nil || placement.ReplicaScheduling.ReplicaSchedulingType == policyv1alpha1.ReplicaSchedulingTypeDuplicated {
+				// reserve unready clusters
+				klog.Warningf("reserve unready clusters due to insufficient available clusters")
+
+				// this is usually 1, which could be more than 1 in some corner cases
+				reservedNum := unreadyBindClusters.Len() - len(candidateClusters)
+				if reservedNum > unreadyBindClusters.Len() {
```

Comment:
unreadyBindClusters.Len() must be > reservedNum

Reply:
Oh yes, my bad